File: backend/src/main/java/t/lab/guide/service/mock/MockPointCategoryService.java
package t.lab.guide.service.mock;

import org.springframework.context.annotation.Profile;
import org.springframework.stereotype.Service;
import t.lab.guide.dto.admin.category.AdminCreateCategoryRequest;
import t.lab.guide.dto.admin.category.AdminPatchCategoryRequest;
import t.lab.guide.dto.point.category.CategoryItem;
import t.lab.guide.dto.point.category.CategoryListResponse;
import t.lab.guide.exception.NotFoundException;
import t.lab.guide.service.PointCategoryService;

import java.util.List;

@Service
@Profile("demo")
public class MockPointCategoryService implements PointCategoryService {

    private static final List<CategoryItem> categories = List.of(
            new CategoryItem(1L, "Достопримечательности", "attraction"),
            new CategoryItem(2L, "Шопинг", "shop"),
            new CategoryItem(3L, "Музеи", "museum"),
            new CategoryItem(4L, "Парки", "park"),
            new CategoryItem(5L, "Еда", "food")
    );

    @Override
    public CategoryListResponse getAllCategories() {
        return new CategoryListResponse(categories);
    }

    @Override
    public CategoryItem createCategory(AdminCreateCategoryRequest request) {
        long newId = categories.size() + 1L;
        return new CategoryItem(newId, request.name(), request.slug());
    }

    @Override
    public CategoryItem patchCategory(Long id, AdminPatchCategoryRequest request) {
        CategoryItem current = categories.stream()
                .filter(c -> c.id().equals(id))
                .findFirst()
                .orElseThrow(() -> new NotFoundException("Категория с id=" + id + " не найдена"));
        return new CategoryItem(
                current.id(),
                request.name() != null ? request.name() : current.name(),
                request.slug() != null ? request.slug() : current.slug()
        );
    }

    @Override
    public void deleteCategory(Long id) {
        categories.stream()
                .filter(c -> c.id().equals(id))
                .findFirst()
                .orElseThrow(() -> new NotFoundException("Категория с id=" + id + " не найдена"));
    }
}

File: backend/src/main/kotlin/t/lab/guide/service/mock/MockPointCategoryService.kt
package t.lab.guide.service.mock

import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Service
import t.lab.guide.dto.admin.category.AdminPatchCategoryRequest
import t.lab.guide.dto.admin.category.command.AdminCreateCategoryCommand
import t.lab.guide.dto.category.CategoryItem
import t.lab.guide.dto.category.CategoryListResponse
import t.lab.guide.exception.NotFoundException
import t.lab.guide.service.PointCategoryService

@Service
@Profile("demo")
class MockPointCategoryService : PointCategoryService {
    override fun getAllCategories(): CategoryListResponse = CategoryListResponse(categories)

    override fun createCategory(request: AdminCreateCategoryCommand): CategoryItem {
        val newId = categories.size + 1L
        return CategoryItem(newId, request.name, request.slug)
    }

    override fun patchCategory(
        id: Long,
        request: AdminPatchCategoryRequest,
    ): CategoryItem {
        val current =
            categories.firstOrNull { it.id == id }
                ?: throw NotFoundException("Категория с id=$id не найдена")
        return CategoryItem(
            id = current.id,
            name = request.name ?: current.name,
            slug = request.slug ?: current.slug,
        )
    }

    override fun deleteCategory(id: Long) {
        categories.firstOrNull { it.id == id }
            ?: throw NotFoundException("Категория с id=$id не найдена")
    }

    companion object {
        private val categories =
            listOf(
                CategoryItem(1L, "Достопримечательности", "attraction"),
                CategoryItem(2L, "Шопинг", "shop"),
                CategoryItem(3L, "Музеи", "museum"),
                CategoryItem(4L, "Парки", "park"),
                CategoryItem(5L, "Еда", "food"),
            )
    }
}
